attestation-service/verifier/src/se/README.md

```diff
@@ -167,10 +179,207 @@ services:
+      - ./data/rsa/encrypt_key.pub:/run/confidential-containers/ibmse/rsa/encrypt_key.pub
+    depends_on:
+    - rvps
+
+  rvps:
+    image: ghcr.io/confidential-containers/staged-images/rvps:latest
+    restart: always # keep the server running
+    ports:
+      - "50003:50003"
+    volumes:
+      - ./data/reference-values:/opt/confidential-containers/attestation-service/reference_values:rw
+      - ./rvps.json:/etc/rvps.json:rw
+EOF
+```
+
+- Prepare rvps.json
+```bash
+cat << EOF > rvps.json
+{
+    "address": "0.0.0.0:50003",
+    "store_type": "LocalFs",
+    "store_config": {
+        "file_path": "/opt/confidential-containers/attestation-service/reference_values"
+    }
+}
+EOF
+```
+
+- Prepare the material, similar as:
+```
+.
+├── data
+│   ├── attestation-service
+│   ├── certs
+│   │   ├── DigiCertCA.crt
+│   │   └── ibm-z-host-key-signing-gen2.crt
+│   ├── crls
+│   │   └── ibm-z-host-key-gen2.crl
+│   ├── hdr
+│   │   └── hdr.bin
+│   ├── hkds
+│   │   └── HKD-3931-0275D38.crt
+│   ├── kbs-storage
+│   │   ├── default
+│   │   └── one
+│   │       └── two
+│   │           └── key
+│   ├── reference-values
+│   └── rsa
+│       ├── encrypt_key.pem
+│       └── encrypt_key.pub
+├── kbs-config.toml
+├── kbs.key
+├── kbs.pem
+└── rvps.json
+```
+
+- Launch KBS + rvps as docker compose application
+```bash
 docker-compose up -d
-docker-compose logs web
+docker-compose logs kbs
 docker-compose down
 ```
 
+- Set reference values into rvps
+
+1. Build rvps client tool
+```bash
+apt-get install protobuf-compiler make gcc -y
+cd ${SRC_ROOT}/trustee/attestation-service/rvps
+make build && make install
+```
+
+2. Edit a test message for IBM SE evidence:
+```bash
+cat << EOF > se-sample
+{
+    "se.attestation_phkh": [
+        "xxx"
+    ],
+    "se.tag": [
+        "xxx"
+    ],
+    "se.image_phkh": [
+        "xxx"
+    ],
+    "se.user_data": [
+        "00"
+    ],
+    "se.version": [
+        "256"
+    ]
+}
+EOF
+provenance=$(cat se-sample | base64 --wrap=0)
+cat << EOF > se-message
+{
+    "version" : "0.1.0",
+    "type": "sample",
+    "payload": "$provenance"
+}
+EOF
+```
+
+Where the values come from [retrive-the-rvps-field-for-an-ibm-se-image](#retrive-the-rvps-field-for-an-ibm-se-image)
+
+3. Register the provenance into rvps with rvps-tool
+```bash
+RVPS_ADDR=127.0.0.1:50003
+rvps-tool register --path ./se-message --addr http://$RVPS_ADDR
+```
+
+- Set attestation policy
+
+1. Generate attestation policy file
+```bash
+cat << EOF > ibmse-policy.rego
+package policy
+import rego.v1
+default allow = false
+
+converted_version := sprintf("%v", [input["se.version"]])
+
+allow if {
+    input["se.attestation_phkh"] == data.reference["se.attestation_phkh"][_]
+    input["se.image_phkh"] == data.reference["se.image_phkh"][_]
+    input["se.tag"] == data.reference["se.tag"][_]
+    input["se.user_data"] == data.reference["se.user_data"][_]
+    converted_version == data.reference["se.version"][_]
```

Comment:
Actually we could directly set the hexs calculated by `parse_hdr.py` on the right side of the `==` here without interact with RVPS right? Current RVPS is not stable and would change frequently, thus I suggest use AS policy only. Or, it might cause a lot of document changes when RVPS changes, which would easily be ignored and thus leaving a bunch of out-of-date documents.

Reply:
Yeah, I know, we actually tried set the value directly, I'm fine to set the value here instead reference value as 1st option. Thanks!

Reply:
Refined, thanks! @Xynnn007